File: 본문/c/Code_3_01_2.c
#include <stdio.h>
#include <stdbool.h>

// 2 이상의 정수 N을 매개변수로 받고, N이 소수라면 True를,
// 아니라면 False를 리턴하는 함수
bool isprime(long long N) {
	long long i;
	for (i = 2; i * i <= N; i++) {
		if (N % i == 0) {
			return false;
		}
	}
	return true;
}

int main() {
	long long N;
	scanf("%lld", &N);
	if (isprime(N) == true) {
		printf("prime\n");
	}
	else {
		printf("not prime\n");
	}
	return 0;
}

File: 본문/python/Code_3_01_2.py
# 2 이상의 정수 N을 매개변수로 받고, N이 소수라면 True를,
# 아니라면 False를 리턴하는 함수
def isprime(N):
	LIMIT = int(N ** 0.5)
	for i in range(2, LIMIT + 1):
		if N % i == 0:
			return False
	return True

N = int(input())
if isprime(N):
	print("prime")
else:
	print("not prime")
